### PR TITLE
Immediate ExpandAll

### PR DESCRIPTION
**Pull request by adamsitnik**
Whenever I clicked ExpandAll the app was freezing and it took a lot of time to do the job.

I used PerfView to profile itself and it turned out, that by default the expanded node was being selected after expanded. This envolved scroll of the grid, which took a lot of time.

expandall flamegraph1

The solution: don't select nodes when Expanding All.

Result:

expandallafter

### CODE REVIEW

File: src/PerfView/StackViewer/CallTreeView.cs [hunk 1]
```diff
@@ -324,49 +324,63 @@ public bool IsExpanded
...
-                    // set the selected node to my caller (if available) or myself if there is none.  
-                    m_treeView.Select(Data.Caller ?? Data);  
-                }
-                ValidateTree();
+            ValidateTree();
+        }
+
+        private void CollapseNode()
+        {
+            ValidateTree();
```
Comment (vancem) [inline src/PerfView/StackViewer/CallTreeView.cs]: Can you put a check (!m_isExpanded) return  at the start to be symmetric with your ExpandNode routine.    Seems safer now that others might call this from locations where the node happened to already be collapsed.

File: src/PerfView/StackViewer/CallTreeView.cs [hunk 2]
```diff
@@ -481,16 +495,17 @@ public string InclusiveMetricByScenarioString
...
                 return;
             if (!expandGraphNodes && Data.IsGraphNode)
                 return;
 
-            IsExpanded = true;
+            ExpandNode(selectExpandedNode);
+
             foreach (var child in VisibleChildren)
-                child.ExpandToDepth(maxDepth - 1, expandGraphNodes);
+                child.ExpandToDepth(maxDepth - 1, expandGraphNodes, selectExpandedNode);
```
Comment (vancem) [inline src/PerfView/StackViewer/CallTreeView.cs]: I note that previously the last node was selected, but now no node is selected.    I guess I am OK with no nodes being selected on expand-all so this is OK (nothing to do).

**Comment by vancem** [comment]
One minor point.  Otherwise looks good to me.

**Comment by adamsitnik** [comment]
@vancem done! thanks for a fast review!

**Comment by vancem** [comment]
The fast review was by luck.   I happened to be trying to work off the backlog.  

Thanks for the contribution.

**Pull request merged**